lib/install.mjs

```diff
@@ -0,0 +1,132 @@
+import { createWriteStream, existsSync, rmSync, renameSync } from 'node:fs';
+import { get } from 'node:https';
+import { dirname, resolve } from 'node:path';
+import { arch, env, platform, exit } from 'node:process';
+import { fileURLToPath, URL } from 'node:url';
+
+import compressing from 'compressing';
+import progress from 'cli-progress';
+import semver from 'semver';
+
+import nodeManifest from "../package.json" assert { type: "json" };
+
+/**
+ * NW.js build flavor
+ * 
+ * @type {'sdk' | 'normal'}
+ */
+let buildType = env.npm_config_nwjs_build_type || env.NWJS_BUILD_TYPE || 'normal';
+
+// Parse Node manifest version.
+let parsedVersion = semver.parse(nodeManifest.version);
+let versionString = `${parsedVersion.major}.${parsedVersion.minor}.${parsedVersion.patch}`;
+
+// Check if version is a prelease.
+if (parsedVersion.prerelease !== undefined && typeof parsedVersion.prerelease[0] === 'string') {
+  let prerelease = parsedVersion.prerelease[0].split('-');
+  if (prerelease.length > 1) {
+    prerelease = prerelease.slice(0, -1);
+  }
+  versionString = `${versionString}-${prerelease}`;
+}
+
+// Check build flavor
+if (versionString.slice(-4) === '-sdk') {
+  versionString = versionString.slice(0, -4);
+  buildType = 'sdk';
+} else if (versionString.slice(-3) === 'sdk') {
+  versionString = versionString.slice(0, -3);
+  buildType = 'sdk';
+}
+
+let url = '';
+let hostArch = env.npm_config_nwjs_process_arch || arch;
+let urlBase = env.npm_config_nwjs_urlbase || env.NWJS_URLBASE || 'https://dl.nwjs.io/v';
+let buildTypeSuffix = buildType === 'normal' ? '' : `-${buildType}`;
+let hostOs = env.npm_config_nwjs_platform || env.NWJS_PLATFORM || platform;
+
+// Determine download url
+switch (hostOs) {
+  case 'win32':
+    url = `${urlBase}${versionString}/nwjs${buildTypeSuffix}-v${versionString}-win-${hostArch}.zip`;
+    break;
+  case 'darwin':
+    url = `${urlBase}${versionString}/nwjs${buildTypeSuffix}-v${versionString}-osx-${hostArch}.zip`;
+    break;
+  case 'linux':
+    url = `${urlBase}${versionString}/nwjs${buildTypeSuffix}-v${versionString}-linux-${hostArch}.tar.gz`;
+    break;
+}
+
+if (url === '') {
+  console.error('[ERROR] Could not find a compatible version of nw.js to download for your platform.');
```

Comment:
In the original code this error would result in a process exit, where as a `console.error` (unlike `throw`) would continue execution.
```suggestion
  console.error('[ERROR] Could not find a compatible version of nw.js to download for your platform.');
  exit(1);
```